pkg/configuration/router.go

```diff
@@ -62,3 +100,114 @@ func updateOrganizationConfig(logger log.Logger, repo Repository) http.HandlerFu
+			moovhttp.Problem(w, fmt.Errorf("problem reading logo file - error=%v", err))
+			return
+		}
+
+		w.Header().Set("Content-Type", rdr.ContentType())
+		w.WriteHeader(http.StatusOK)
+		w.Write(fBytes)
+	}
+}
+
+func uploadOrganizationLogo(logger log.Logger, repo Repository, bucketFactory storage.BucketFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		requestID := moovhttp.GetRequestID(r)
+		organization := route.GetOrganization(w, r)
+		if organization == "" {
+			logger.Log("configuration", "upload logo called with no organization header", "requestID", requestID)
+			return
+		}
+
+		file, _, err := r.FormFile("file")
+		if file == nil || err != nil {
+			logger.Log("configuration", errMissingFile, "error", err, "organization", organization, "requestID", requestID)
+			moovhttp.Problem(w, errMissingFile)
+			return
+		}
+		defer file.Close()
+
+		// Detect the content type by reading the first 512 bytes of r.Body (read into file as we expect a multipart request)
+		buf := make([]byte, 512)
+		if _, err := file.Read(buf); err != nil && err != io.EOF {
+			logger.Log("configuration", "problem reading file", "error", err, "organization", organization, "requestID", requestID)
+			moovhttp.Problem(w, err)
+			return
+		}
+
+		contentType := http.DetectContentType(buf)
+		if !allowedContentTypes[contentType] {
+			logger.Log("configuration", "unsupported content type for logo image file", "error", err, "contentType", contentType, "organization", organization, "requestID", requestID)
+			moovhttp.Problem(w, errUnsupportedType)
+			return
+		}
+
+		bucket, err := bucketFactory()
+		if err != nil {
+			logger.Log("configuration", "problem uploading logo image", "error", err, "organization", organization, "requestID", requestID)
+			moovhttp.Problem(w, err)
+			return
+		}
+		defer bucket.Close()
+
+		ctx, cancelFn := context.WithTimeout(r.Context(), 60*time.Second)
+		defer cancelFn()
+
+		writer, err := bucket.NewWriter(ctx, makeDocumentKey(organization), &blob.WriterOptions{
+			ContentDisposition: "inline",
+			ContentType:        contentType,
+		})
+		if err != nil {
+			logger.Log("configuration", "problem uploading logo image", "error", err, "organization", organization, "requestID", requestID)
+			moovhttp.Problem(w, err)
+			return
+		}
+		defer writer.Close()
+
+		written, err := io.Copy(writer, io.LimitReader(io.MultiReader(bytes.NewReader(buf), file), maxImageSize))
+		if err != nil || written == 0 {
+			logger.Log("configuration", "problem uploading logo image", "error", err, "organization", organization, "requestID", requestID)
+			moovhttp.Problem(w, fmt.Errorf("problem writing file - wrote %d bytes with error=%v", written, err))
+			return
+		}
+
+		w.WriteHeader(http.StatusNoContent)
+	}
+}
+
+func makeDocumentKey(organization string) string {
+	return path.Join("configuration", organization, "logo")
```

Comment:
I'm wondering if this path should be `organizations/{organizationID}/logo` instead? That path would be a bit clearer.

Reply:
That would make sense. I'll make the change.